pennylane/interfaces/autograd.py

```diff
@@ -258,17 +267,264 @@ def grad_fn(dy):
+            # No additional quantum evaluations needed; simply compute the VJPs directly.
+            vjps = _compute_vjps(jacs, dy, multi_measurements)
+
+        else:
+            # Need to compute the Jacobians on the backward pass (accumulation="backward")
+            if isinstance(gradient_fn, qml.gradients.gradient_transform):
+                # Gradient function is a gradient transform.
+
+                # Generate and execute the required gradient tapes
+                if _n == max_diff:
+                    with qml.tape.Unwrap(*tapes):
+                        vjp_tapes, processing_fn = qml.gradients.batch_vjp(
+                            tapes,
+                            dy,
+                            gradient_fn,
+                            reduction="append",
+                            gradient_kwargs=gradient_kwargs,
+                        )
+
+                        vjps = processing_fn(execute_fn(vjp_tapes)[0])
+
+                else:
+                    vjp_tapes, processing_fn = qml.gradients.batch_vjp(
+                        tapes, dy, gradient_fn, reduction="append", gradient_kwargs=gradient_kwargs
+                    )
+
+                    # This is where the magic happens. Note that we call ``execute``.
+                    # This recursion, coupled with the fact that the gradient transforms
+                    # are differentiable, allows for arbitrary order differentiation.
+                    vjps = processing_fn(
+                        _execute_new(
+                            vjp_tapes,
+                            device,
+                            execute_fn,
+                            gradient_fn,
+                            gradient_kwargs,
+                            _n=_n + 1,
+                            max_diff=max_diff,
+                        )
+                    )
+
+            else:
+                # Gradient function is not a gradient transform
+                # (e.g., it might be a device method).
+                # Note that unlike the previous branch:
+                #
+                # - there is no recursion here
+                # - gradient_fn is not differentiable
+                #
+                # so we cannot support higher-order derivatives.
+                with qml.tape.Unwrap(*tapes):
+                    jacs = gradient_fn(tapes, **gradient_kwargs)
+
+                vjps = _compute_vjps(jacs, dy, multi_measurements)
+
+        return_vjps = [
+            qml.math.to_numpy(v, max_depth=_n) if isinstance(v, ArrayBox) else v for v in vjps
+        ]
+        if device.short_name == "strawberryfields.gbs":  # pragma: no cover
+            # TODO: remove this exceptional case once the source of this issue
+            # https://github.com/PennyLaneAI/pennylane-sf/issues/89 is determined
+            return (return_vjps,)  # pragma: no cover
+        return return_vjps
+
+    return grad_fn
+
+
+def _compute_vjps(jacs, dy, multi_measurements):

```

Comment:
It is only used by the interfaces when we use a device method a gradient function (vjp contain mostly function for gradient transform). If @eddddddy needs for Tensorflow, we should probably move it and fine a new name like `compute_vjps_device`

Reply:
Maybe good to add "autograd" to the function name for now. My worry is that the current name is quite general and remaining as is may make the readability of the code a challenge (since there are very similarly named functions in the `gradients` module).

Reply:
For TensorFlow the function is similar but slightly different (we unstack the results and extend the list). I think having separate functions for each interface is fine as long as it's clear that they are interface-specific.

Reply:
I will a suffix then!